src/resolve/resolved-dns-query.c

```diff
@@ -585,6 +586,13 @@ void dns_query_complete(DnsQuery *q, DnsTransactionState state) {
 
         q->state = state;
 
+        if (state == DNS_TRANSACTION_SUCCESS && set_size(q->manager->varlink_subscription) > 0) {
+                DnsQuestion *question = q->request_packet ? q->request_packet->question : NULL;
+                const char *query_name = question ? dns_question_first_name(question) : q->request_name;
+                if (query_name)
+                        send_dns_notification(q->manager, q->answer, query_name);
```

Comment:
I requested to voidify the return value, but the function itself should return a negative errno on failure:
```c
(void) send_dns_notification(...);

int send_dns_notification(...) {
    ...
    if (r < 0)
        return log_debug_errno(r, "Failed to ...");
}
```

Reply:
Got it, thanks. I misunderstood the earlier comment. Updated now to return an error again